getnacelle/nacelle-js
# chore(vue): update to vue 2.7

- feat: replace vue-composition-api with vue 2.7
- chore: update to vite v3 & vue-2.7 vite plugin
- chore: update to use latest @nuxtjs-composition-api
- docs: adds changeset



## Why are these changes introduced?

Addresses [ENG-8369](https://nacelle.atlassian.net/browse/ENG-8369) -  make `@nacelle/vue` compatible with `@nuxtjs/composition-api` 0.33+ 



## What is this pull request doing?

1. Updates `@nacelle/vue` to replace `@vue/composition-api` with vue 2.7 & updates tests accordingly (vue 2.7 unwraps refs on, the vm object, which was not happening with `@vue/composition-api`, breaking a handful of tests).
2. Updates `@nacelle/vue` to vite 3 to use the `@vitejs/plugin-vue2` which is recommended for vite + vue 2.7.
3. Updates `@nacelle/vue` to use `jest@29`
4. Updates `examples/nuxt-2-composition-api` to use `@nuxtjs/composition-api@^0.33` & updates the underlying Nuxt version to use vue 2.7.


## How to Test


1. Checkout this branch `mdarrik/ENG-8369-update-nacelle-vue-to-vue-2-7`
2. From the root of the repo run `npm run bootstrap` 
3. Navigate to `packages/vue`
4. Run `npm run test` all tests should pass.
5. Run `npm run build` build should succeed
6. Run `npm link`
7. Navigate to `examples/nuxt-2-composition-api`
8. Make sure your `.env` file is setup with a Nacelle v1 space. 
9. Run `npm link @nacelle/vue` 
10. Run `npm run generate` - should generate your pages successfully. 
11. Run `npx serve dist/` - should be able to navigate to products/collections pages that render correctly


## Checklist



[ENG-8369]: https://nacelle.atlassian.net/browse/ENG-8369?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
